packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -26,6 +26,8 @@ import 'text_span.dart';
 
 export 'package:flutter/services.dart' show TextRange, TextSelection;
 
+const Object _debugAssert = Object();
```

Comment:
Why does every file have to define this anew? Why not define this one somewhere in `foundation`? e.g. `foundation/debug.dart`.

Reply:
Does foundation get re-exported by higher-level libraries? This is for framework code only so I'd like to keep it private to `packages/flutter/lib`

Reply:
Foundation is available to other developers, but I don't think it matters. We could take the style of package:meta and just document that tools could use this annotation to verify xyz. It's not guaranteed that general purpose tools do, though. 

If you look at foundation/annotations.dart you'll find that we have other custom annotations in there as well that only make sense for flutter-specific tooling.